src/services/modules/rest/impl/src/main/java/org/geoserver/geofence/services/rest/impl/RESTRuleServiceImpl.java

```diff
@@ -125,6 +127,49 @@ public Response insert(RESTInputRule inputRule) throws NotFoundRestEx, BadReques
         }
     }
 
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED, value = "geofenceTransactionManager")
```

Comment:
same as above. The @Transactional annotation should not go over the service level. Please move it either to the service insertion level or to the DAO one.